lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -300,18 +300,22 @@ function dosomething_signup_exists($nid, $runId = NULL, $uid = NULL, $presignup
     $uid = $user->uid;
   }
 
-  $signup_tbl = 'dosomething_signup';
-
   if ($presignup) {
-    $signup_tbl = 'dosomething_signup_presignup';
+    $result = db_select('dosomething_signup_presignup', 's')
+      ->condition('uid', $uid)
+      ->condition('nid', $nid)
+      ->fields('s', array('sid'))
+      ->execute();
+  }
+  else {
+    $result = db_select('dosomething_signup', 's')
+      ->condition('uid', $uid)
+      ->condition('nid', $nid)
+      ->condition('run_nid', $runId)
+      ->fields('s', array('sid'))
+      ->execute();
   }
 
-  $result = db_select($signup_tbl, 's')
-    ->condition('uid', $uid)
-    ->condition('nid', $nid)
-    ->condition('run_nid', $runId)
-    ->fields('s', array('sid'))
-    ->execute();
   $sid = $result->fetchField(0);
```

Comment:
First, you need to check `if (!is_null($result))`, as [execute](https://api.drupal.org/api/drupal/includes!database!select.inc/function/SelectQuery%3A%3Aexecute/7) method may return `NULL`.